spec/models/topic_spec.rb

```diff
@@ -3095,6 +3106,18 @@ def set_state!(group, user, state)
         expect(post.post_type).to eq(Post.types[:small_action])
         expect(post.action_code).to eq("user_left")
       end
+
+      it "should show a small action when user removes themselves from pm" do
+        create_post(topic: private_topic, user: admin)
+        create_post(topic: private_topic, user: user1)
+
+        private_topic.remove_allowed_user(user1, user1)
+        expect(private_topic.allowed_users.include?(user1)).to eq(false)
+
+        posts = private_topic.posts
+        expect(posts.size).to eq(3)
+        expect(posts.last.action_code).to eq("user_left")
```

Comment:
```suggestion
        expect do
          private_topic.remove_allowed_user(user1, user1)
        end.to change { private_topic.posts.where(action_code: "user_left").count }.by(1)
```

What do you think about just asserting that a small action post was created? I feel like the other smaller assertions are adding much value for what this test was created for.